File: petties_mobile/lib/data/models/vaccine_template.dart
class VaccineTemplate {
  final String id;
  final String name;
  final String? manufacturer;
  final String? description;
  final int? minAgeWeeks;
  final int? repeatIntervalDays;
  final int? seriesDoses;
  final bool? isAnnualRepeat;
  final double? defaultPrice;
  final String targetSpecies;
  final int? minIntervalDays;

  VaccineTemplate({
    required this.id,
    required this.name,
    this.manufacturer,
    this.description,
    this.minAgeWeeks,
    this.repeatIntervalDays,
    this.seriesDoses,
    this.isAnnualRepeat,
    this.defaultPrice,
    required this.targetSpecies,
    this.minIntervalDays,
  });

  factory VaccineTemplate.fromJson(Map<String, dynamic> json) {
    return VaccineTemplate(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      manufacturer: json['manufacturer'],
      description: json['description'],
      minAgeWeeks: json['minAgeWeeks'],
      repeatIntervalDays: json['repeatIntervalDays'],
      seriesDoses: json['seriesDoses'],
      isAnnualRepeat: json['isAnnualRepeat'],
      defaultPrice: (json['defaultPrice'] as num?)?.toDouble(),
      targetSpecies: json['targetSpecies'] ?? 'BOTH',
      minIntervalDays: json['minIntervalDays'],
    );
  }
}

File: petties-web/src/services/endpoints/vaccineTemplate.ts
import { apiClient } from '../api/client'

export interface VaccineTemplate {
    id: string
    name: string
    manufacturer?: string
    description?: string
    minAgeWeeks?: number
    repeatIntervalDays?: number
    minIntervalDays?: number
    seriesDoses?: number
    isAnnualRepeat?: boolean
    defaultPrice?: number
    targetSpecies: 'DOG' | 'CAT' | 'BOTH'
}

export const getAllVaccineTemplates = async (): Promise<VaccineTemplate[]> => {
    const response = await apiClient.get('/vaccine-templates')
    return response.data
}

export const getVaccineTemplateById = async (id: string): Promise<VaccineTemplate> => {
    const response = await apiClient.get(`/vaccine-templates/${id}`)
    return response.data
}
